### DISCUSSION TITLE
Getting basic "Hello Triangle" sample to build and run on M1-M2 mac to get started

### DISCUSSION THREAD

**Discussion by peterkennard** [KhronosGroup/MoltenVK#1844] [Q&A]
I have tried downloading the Vulkan SDK "prebuilt" latest on vulkan.lunarg.com -  The binary executables in it run.

I got a few samples to try and build but they all failed some with link errors saying the .dylib and the compiled app were on incompatible platforms ( arm vs x86 etc ) 

I am new to XCode and a Cmake with a known to work config would be great but ...  I am familiar with "Modern Open GL"

->  do I need to build MoltenVK from source for the M! and/or M2 ?  or is the prebuilt known to work building on M1 - M2

Once done what is the simplest least indirected and factored C++ sample that is known to work on an M1-M2 should I start with.  Ideally with complete instructions for how to get it up on M1-M2.  Maybe a simple spinning cube or a triangle.

Thanks.

**Comment by billhollings** [comment]
> I got a few samples to try and build but they all failed some with link errors saying the .dylib and the compiled app were on incompatible platforms ( arm vs x86 etc )

Try posting this question to the [Vulkan SDK issue list](https://vulkan.lunarg.com/issue/home) regarding the platform builds included in the SDK.

> what is the simplest least indirected and factored C++ sample that is known to work on an M1-M2 should I start with. Ideally with complete instructions for how to get it up on M1-M2. Maybe a simple spinning cube or a triangle.

There is a Cube Demo included in MoltenVK. Once you clone from this repo, it is very straightforward to build both MoltenVK and the Cube Demo. You can build from command line, but you will need Xcode.

**Comment by H-G-Hristov** [comment]
Is it a good idea to use MoltenVK and macOS as a primary platform for development or is it better to do it on Windows and just port to macOS? Is there anything missing on macOS compared to developing on Windows?

**Comment by peterkennard** [comment]
Well the validation layer and the debug message logger are helpful and not
available on .moltenVK

**Reply by cdavis5e** [reply]
> Well the validation layer and the debug message logger are helpful and not available on .moltenVK

Only if you link directly to MoltenVK. If you link to the loader and use MoltenVK as an ICD, you get all that, just like you would on Windows or Linux.
